### PR TITLE
[WIP] Feature/translate into typescript

### PR DESCRIPTION
**Pull request by sihu**

### CODE REVIEW

File: src/renuo_upload.ts [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+  private checkAdaptParams() {
+    this.checkElement();
+    this.adaptCallback();
+    this.adaptOptions();
+  }
+
+  private checkRequirements() {
+    if (typeof jQuery === 'undefined' || jQuery === null)
+      throw new Error('RenuoUpload needs jQuery.');
```
Comment (lukaselmer) [inline src/renuo_upload.ts]: one-liner or {} (i would prefer a one-liner)

File: src/renuo_upload.ts [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+  }
+
+  private adaptCallback() {
+    if (typeof this.callback !== 'function')
+      this.callback = this.defaultCallback;
+  }
+
+  private defaultCallback(result:RenuoUploadResult) {
+    if (jQuery(this.element).parents('form').length) {
+      name = result.name
```
Comment (lukaselmer) [inline src/renuo_upload.ts]: ;

File: src/renuo_upload.ts [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+      //TODO discuss with Lukas about upload of two times the same file
+      delete result.name
```
Comment (lukaselmer) [inline src/renuo_upload.ts]: ;

Comment (schmijos) [inline src/renuo_upload.ts]: This smells fishy. Why using [`delete`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/delete)?

Comment (lukaselmer) [inline src/renuo_upload.ts]: @schmijos yes indeed. tbd with @cyrilkyburz, so we can remove the todos (i'm pretty sure that @cyrilkyburz and i discussed it already, but it was some time ago...)

**Pull request closed without merging**